modules/oncodrivefml/1.0/oncodrivefml.smk

```diff
@@ -0,0 +1,252 @@
+
+checkpoint _oncodrivefml_prep_input:
+    input:
+        maf = expand(
+            str(rules._oncodrivefml_input_maf.output.maf),
+            seq_type = CFG["samples"]["seq_type"].unique(),
+            allow_missing=True),
+        subsetting_categories = ancient(str(rules._oncodrivefml_sample_set.output.subsetting_categories))
+    output:
+        CFG["dirs"]["inputs"] + "maf/{genome_build}/{sample_set}--{launch_date}/done"
+    log:
+        stdout = CFG["logs"]["inputs"] + "{genome_build}/{sample_set}--{launch_date}/prep_input/prep_input_maf.log"
+    conda:
+        CFG["conda_envs"]["prepare_mafs"]
+    params:
+        include_non_coding = str(CFG["maf_processing"]["include_non_coding"]).upper(),
+        mode = "OncodriveFML",
+        metadata_cols = CFG["samples"],
+        metadata = CFG["samples"].to_numpy(na_value='')
+    script:
+        PREPARE_MAFS
+    
+rule _oncodrivefml_blacklist:
+    input:
+        maf = CFG["dirs"]["inputs"] + "maf/{genome_build}/{sample_set}--{launch_date}/{md5sum}.maf",
+        blacklists = CFG["maf_processing"]["blacklists"],
+        deblacklist_script = CFG["scripts"]["deblacklist_script"]
+    output:
+        maf = CFG["dirs"]["inputs"] + "maf/{genome_build}/{sample_set}--{launch_date}/{md5sum}.deblacklisted.maf"
+    params:
+        drop_threshold = CFG["maf_processing"]["blacklist_drop_threshold"]
+    log:
+        stdout = CFG["logs"]["inputs"] + "{genome_build}/{sample_set}--{launch_date}/{md5sum}/deblacklist/deblacklist.stdout.log",
+        stderr = CFG["logs"]["inputs"] + "{genome_build}/{sample_set}--{launch_date}/{md5sum}/deblacklist/deblacklist.stderr.log"
+    shell:
+        op.as_one_line("""
+        {input.deblacklist_script}
+        --input {input.maf}
+        --output {output.maf}
+        --drop-threshold {params.drop_threshold}
+        --blacklists {input.blacklists}
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+rule _oncodrivefml_format_input:
+    input:
+        maf = str(rules._oncodrivefml_blacklist.output.maf)
+    output:
+        maf = CFG["dirs"]["inputs"] + "maf/{genome_build}/{sample_set}--{launch_date}/{md5sum}.fml_input.maf"
+    params:
+        columns = CFG["format_fml_input"]["input_columns"],
+        additional_commands = CFG["format_fml_input"]["additional_commands"]
+    shell:
+        op.as_one_line("""
+        cut -f {params.columns} {input.maf} |
+        sed 's/Chromosome/CHROMOSOME/' |
+        sed 's/Start_Position/POSITION/' |
+        sed 's/Reference_Allele/REF/' |
+        sed 's/Tumor_Seq_Allele2/ALT/' |
+        sed 's/Tumor_Sample_Barcode/SAMPLE/'
+        {params.additional_commands}
+        > {output.maf}
+        """)
+
+ONCODRIVE_BUILD_DICT = {
+    'hg38': 'hg38',
+    'grch37': 'hg19'
+}
+
+rule _oncodrivefml_get_hg19_scores:
+    output:
+        scores = CFG["dirs"]["cadd"] + "grch37/genomicscores/caddpack/1.0-20170217/whole_genome_SNVs.fml",
```

Comment:
is this hard-coded `grch37`, do we need to wildcard this?

Reply:
It is hard coded because only the hg19 scores are available through the bgdata framework provided by bbglab